### ISSUE TITLE
the server doesn't have a resource type "dc"

### ISSUE THREAD

**Issue by sosiouxme** [openshift/origin#17915]
`oc` from master does not allow me to scale a DC. When I first hit this it gave me the message `the server doesn't have a resource type "dc"`. Now it says `error: Scaling the resource failed with: the API version in the data (extensions/v1beta1) does not match the expected API version (apps.openshift.io/v1); Current resource version Unknown`.

##### Version

```
$ minishift version
minishift v1.10.0+10461c6
$ oc version
oc v3.9.0-alpha.0+849a131-277-dirty
kubernetes v1.9.0-beta1
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://192.168.42.213:8443
openshift v3.7.0+7ed6862
kubernetes v1.7.6+a08f5eeb62
```

##### Steps To Reproduce
1. `minishift start --openshift-version v3.7.0 --memory 4G`
2. ensure `oc` is built from master
3. `oc login -u system:admin`
4. `oc -n default scale --replicas=0 dc/router --loglevel=8`

Update: results are the same using `oc cluster up`

##### Current Result

```
I1221 10:44:32.157542    9314 round_trippers.go:414] GET https://192.168.42.213:8443/oapi/v1/namespaces/default/deploymentconfigs/router
I1221 10:44:32.157559    9314 round_trippers.go:421] Request Headers:
I1221 10:44:32.157565    9314 round_trippers.go:424]     User-Agent: oc/v1.9.0 (linux/amd64) kubernetes/3258431
I1221 10:44:32.157570    9314 round_trippers.go:424]     Accept: application/json, */*
I1221 10:44:32.159703    9314 round_trippers.go:439] Response Status: 200 OK in 2 milliseconds
I1221 10:44:32.159713    9314 round_trippers.go:442] Response Headers:
I1221 10:44:32.159717    9314 round_trippers.go:445]     Cache-Control: no-store
I1221 10:44:32.159721    9314 round_trippers.go:445]     Content-Type: application/json
I1221 10:44:32.159725    9314 round_trippers.go:445]     Content-Length: 3744
I1221 10:44:32.159728    9314 round_trippers.go:445]     Date: Thu, 21 Dec 2017 15:44:32 GMT
I1221 10:44:32.159882    9314 request.go:873] Response Body: {"kind":"DeploymentConfig","apiVersion":"v1","metadata":{"name":"router","namespace":"default","selfLink":"/oapi/v1/namespaces/default/deploymentconfigs/router","uid":"70784d10-e664-11e7-90be-d6886bf10021","resourceVersion":"1262","generation":1,"creationTimestamp":"2017-12-21T15:34:31Z","labels":{"router":"router"}},"spec":{"strategy":{"type":"Rolling","rollingParams":{"updatePeriodSeconds":1,"intervalSeconds":1,"timeoutSeconds":600,"maxUnavailable":"25%","maxSurge":0},"resources":{},"activeDeadlineSeconds":21600},"triggers":[{"type":"ConfigChange"}],"replicas":1,"test":false,"selector":{"router":"router"},"template":{"metadata":{"creationTimestamp":null,"labels":{"router":"router"}},"spec":{"volumes":[{"name":"server-certificate","secret":{"secretName":"router-certs","defaultMode":420}}],"containers":[{"name":"router","image":"openshift/origin-haproxy-router:v3.7.0","ports":[{"hostPort":80,"containerPort":80,"protocol":"TCP"},{"hostPort":443,"containerPort":443,"protocol":"TCP"},{"name":"stats","hostPort":1 [truncated 2720 chars]
I1221 10:44:33.161327    9314 round_trippers.go:414] GET https://192.168.42.213:8443/apis/apps.openshift.io/v1/namespaces/default/deploymentconfigs/router/scale
I1221 10:44:33.161349    9314 round_trippers.go:421] Request Headers:
I1221 10:44:33.161356    9314 round_trippers.go:424]     Accept: application/json, */*
I1221 10:44:33.161361    9314 round_trippers.go:424]     User-Agent: oc/v1.9.0 (linux/amd64) kubernetes/3258431
I1221 10:44:33.163471    9314 round_trippers.go:439] Response Status: 200 OK in 2 milliseconds
I1221 10:44:33.163572    9314 round_trippers.go:442] Response Headers:
I1221 10:44:33.163579    9314 round_trippers.go:445]     Cache-Control: no-store
I1221 10:44:33.163584    9314 round_trippers.go:445]     Content-Type: application/json
I1221 10:44:33.163589    9314 round_trippers.go:445]     Content-Length: 417
I1221 10:44:33.163594    9314 round_trippers.go:445]     Date: Thu, 21 Dec 2017 15:44:33 GMT
I1221 10:44:33.163614    9314 request.go:873] Response Body: {"kind":"Scale","apiVersion":"apps.openshift.io/v1","metadata":{"name":"router","namespace":"default","selfLink":"/apis/apps.openshift.io/v1/namespaces/default/deploymentconfigs/router/scale","uid":"70784d10-e664-11e7-90be-d6886bf10021","resourceVersion":"1262","creationTimestamp":"2017-12-21T15:34:31Z"},"spec":{"replicas":1},"status":{"replicas":1,"selector":{"router":"router"},"targetSelector":"router=router"}}
I1221 10:44:33.163941    9314 request.go:873] Request Body: {"kind":"Scale","apiVersion":"extensions/v1beta1","metadata":{"name":"router","namespace":"default","selfLink":"/apis/apps.openshift.io/v1/namespaces/default/deploymentconfigs/router/scale","uid":"70784d10-e664-11e7-90be-d6886bf10021","resourceVersion":"1262","creationTimestamp":"2017-12-21T15:34:31Z"},"spec":{},"status":{"replicas":1,"selector":{"router":"router"},"targetSelector":"router=router"}}
I1221 10:44:33.163975    9314 round_trippers.go:414] PUT https://192.168.42.213:8443/apis/apps.openshift.io/v1/namespaces/default/deploymentconfigs/router/scale
I1221 10:44:33.163982    9314 round_trippers.go:421] Request Headers:
I1221 10:44:33.163988    9314 round_trippers.go:424]     Accept: application/json, */*
I1221 10:44:33.163993    9314 round_trippers.go:424]     User-Agent: oc/v1.9.0 (linux/amd64) kubernetes/3258431
I1221 10:44:33.165354    9314 round_trippers.go:439] Response Status: 400 Bad Request in 1 milliseconds
I1221 10:44:33.165363    9314 round_trippers.go:442] Response Headers:
I1221 10:44:33.165368    9314 round_trippers.go:445]     Content-Type: application/json
I1221 10:44:33.165371    9314 round_trippers.go:445]     Content-Length: 226
I1221 10:44:33.165374    9314 round_trippers.go:445]     Date: Thu, 21 Dec 2017 15:44:33 GMT
I1221 10:44:33.165378    9314 round_trippers.go:445]     Cache-Control: no-store
I1221 10:44:33.165391    9314 request.go:873] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"the API version in the data (extensions/v1beta1) does not match the expected API version (apps.openshift.io/v1)","reason":"BadRequest","code":400}
F1221 10:44:33.165508    9314 helpers.go:119] error: Scaling the resource failed with: the API version in the data (extensions/v1beta1) does not match the expected API version (apps.openshift.io/v1); Current resource version Unknown
```

**Comment by php-coder** [comment]
CC @tnozicka

**Comment by mfojtik** [comment]
@liggitt PTAL, I thought we fixed the scaling endpoints

**Comment by liggitt** [comment]
fixed in origin 3.7 branch (no 3.7.1 release cut yet): https://github.com/openshift/origin/pull/17517

**Comment by liggitt** [comment]
v3.7.1 released
